src/cryptonote_core/miner.cpp

```diff
@@ -408,5 +528,313 @@ namespace cryptonote
     return true;
   }
   //-----------------------------------------------------------------------------------------------------
-}
+  bool miner::get_is_background_mining_enabled() const
+  {
+    return m_is_background_mining_enabled;
+  }
+  //-----------------------------------------------------------------------------------------------------
+  bool miner::set_is_background_mining_enabled(const bool& is_background_mining_enabled)
+  {
+    if( is_background_mining_enabled && !epee::file_io_utils::is_file_exist("/proc/stat") )
```

Comment:
This is only going to work on systems with this /proc layout ... should the process arguments be `#ifndef` on systems where this is known in advance to not be available (i.e. WIN32, etc.)?